src/server/svr_jobfunc.c

```diff
@@ -703,10 +704,11 @@ svr_setjobstate(job *pjob, int newstate, int newsubstate)
 	/* update the job file */
 
 	if (newstate == JOB_STATE_RUNNING) {
-		set_entity_ct_sum_queued(pjob, NULL, DECR);
-		set_entity_ct_sum_queued(pjob, pjob->ji_qhdr, DECR);
-		set_entity_resc_sum_queued(pjob, NULL, NULL, DECR);
-		set_entity_resc_sum_queued(pjob, pjob->ji_qhdr, NULL, DECR);
+		if (pjob->ji_etlimit_decr_queued == FALSE) {
+			account_entity_limit_usages(pjob, NULL, NULL, DECR, ETLIM_ACC_ALL_QUEUED);
+			account_entity_limit_usages(pjob, pjob->ji_qhdr, NULL, DECR, ETLIM_ACC_ALL_QUEUED);
+			pjob->ji_etlimit_decr_queued = TRUE;
+		}
```

Comment:
Here job is about to enter `JOB_STATE_RUNNING` state, hence we need to decrement all the queued limits. Hence invoke `account_entity_limit_usages()` with `ETLIM_ACC_ALL_QUEUED`.  Also mark `pjob->ji_etlimit_decr_queued` as set so that we wont decrement the queued limits again during job purge.